library/Fission/Web/User/Create.hs

```diff
@@ -31,26 +33,40 @@ withDID ::
   ( MonadDNSLink m
   , MonadLogger  m
   , MonadTime    m
+  , MonadMail    m
   , User.Creator m
+  , Challenge.Creator m
   )
   => DID
   -> ServerT API m
 withDID DID {..} User.Registration {username, email} = do
   now <- currentTime
-  Web.Err.ensureM $ User.create username publicKey email now
-  return NoContent
+  userId <- Web.Err.ensureM $ User.create username publicKey email now
+  challenge <- Web.Err.ensureM $ Challenge.create userId
+  res <- sendVerifyEmail (Recipient email username) challenge
+  case res of
+    Left err -> do
+      logError $ displayShow err
+      Web.Err.throw err500 { errBody = "Could not send verification email" }
+    Right _ -> return NoContent
```

Comment:
Remember that you can also write this as:

```haskell
sendVerifyEmail (Recipient email username) challenge >>= \case
  Left  err -> ...
  Right val -> ...
```

Reply:
yup :+1: 
I like this better, but I thought we were trying to be consistent within functions of either using `do` notation or `bind` notation

Is there a good rule of thing on when we should mix them? I feel like `case` statements lend themselves really nicely to binds

Reply:
I mean, they're totally equivalent. The main difference is that there's a branch point; we're not really using the variable anywhere else and it's mainly for control flow. Really either is fine, but `>>= \case` saves you a line and I find it a bit more expressive, _unless_ you're doing something like normalizing the response and continuing

```haskell
response <- request

normalized <- case response of
  Left  err -> defaultValue
  Right val -> foo val

bar normalized
```

Reply:
Also, a LOT of these `>>= \case`s will probably disappear when we have `rescue` integrated

Reply:
> I feel like case statements lend themselves really nicely to binds

Yes I'd say that's the exception to the rule, and is increasing popular with the community broadly

Reply:
okay great! :ok_hand: